Release/src/http/client/x509_cert_utilities.cpp

```diff
@@ -36,19 +36,31 @@
 #include <wincrypt.h>
 #endif
 
-namespace web { namespace http { namespace client { namespace details {
+#include <iomanip>
 
-static bool verify_X509_cert_chain(const std::vector<std::string> &certChain, const std::string &hostName);
+namespace web { namespace http { namespace client { namespace details {
 
-bool verify_cert_chain_platform_specific(boost::asio::ssl::verify_context &verifyCtx, const std::string &hostName)
+bool is_end_certificate_in_chain(boost::asio::ssl::verify_context &verifyCtx)
 {
     X509_STORE_CTX *storeContext = verifyCtx.native_handle();
     int currentDepth = X509_STORE_CTX_get_error_depth(storeContext);
     if (currentDepth != 0)
+    {
+        return false;
+    }
+        return true;
+}
+
+#if !defined(__linux__)
+
+bool verify_cert_chain_platform_specific(boost::asio::ssl::verify_context &verifyCtx, const std::string &hostName, const CertificateChainFunction& func)
+{
+    if (!is_end_certificate_in_chain(verifyCtx))
     {
         return true;
     }
 
+    X509_STORE_CTX *storeContext = verifyCtx.native_handle();
```

Comment:
Noticed while pulling this into an older code version: This variable appears to be unused, and native_handle() does not (to my knowledge) have side-effects. 
Therefore suggest removing this line.